test/integration/v29_int_test.go

```diff
@@ -0,0 +1,174 @@
+		e2e.WithArgs(script, "-t", ts.Dirs.Work, "-b", "beta"),
+		e2e.AppendEnv(
+			fmt.Sprintf("%s=%s", constants.OverwriteDefaultInstallationPathEnvVarName, filepath.Join(ts.Dirs.Work, "multi-file")),
+		),
+		e2e.AppendEnv(extraEnv...),
+	)
+
+	cp.Expect("proceed with install?")
+	cp.SendLine("Y")
+	cp.Expect(fmt.Sprintf("Fetching the latest version: %s", rcVersion))
+	if runtime.GOOS == "windows" {
+		cp.Expect("successfully installed to", 30*time.Second)
+	} else {
+		cp.Expect("Installing to", 30*time.Second)
+		cp.Expect("Allow $PATH to be appended in your")
+		cp.SendLine("n")
+		cp.Expect("State Tool installation complete")
+	}
+	cp.ExpectExitCode(0)
+
+	stateExe := filepath.Join(ts.Dirs.Work, "state")
+	if runtime.GOOS == "windows" {
+		stateExe = stateExe + ".exe"
+	}
+	suite.compareVersionedInstall(ts, stateExe, rcVersion, suite.Equal)
+
+	return stateExe
+}
+
+// TestTaggedUpdateFlow is meant to test whether installing our release-candidate will update correctly
+// It is supposed to not attempt an auto-update, and the tag should be set
+func (suite *V29TestSuite) TestTaggedUpdateFlow() {
+	suite.OnlyRunForTags(tagsuite.V29Update, tagsuite.Critical)
+
+	ts := e2e.New(suite.T(), true)
+	defer ts.Close()
+
+	stateExe := suite.installReleaseCandidate(ts)
+
+	// ensure tag is in config yaml file
+	f, err := os.ReadFile(filepath.Join(ts.Dirs.Config, "config.yaml"))
+	suite.Require().NoError(err)
+	suite.Require().Contains(string(f), fmt.Sprintf(`tag: %s`, rcTag))
+
+	suite.Run("Tagged RC should not update", func() {
+		cp := ts.SpawnCmdWithOpts(
+			stateExe,
+			e2e.WithArgs("update"),
+			e2e.AppendEnv(
+				fmt.Sprintf("%s=%s", constants.OverwriteDefaultInstallationPathEnvVarName, filepath.Join(ts.Dirs.Work, "multi-file")),
+			),
+		)
+		cp.Expect("You are already using the latest State Tool version")
+		cp.ExpectExitCode(0)
+	})
+
+	suite.Run("Update without tag", func() {
+		// remove update-tag
+		err := os.Remove(filepath.Join(ts.Dirs.Config, "config.yaml"))
+		suite.Require().NoError(err)
+
+		cp := ts.SpawnCmdWithOpts(
+			stateExe,
+			e2e.WithArgs("update"),
+			e2e.AppendEnv(
+				fmt.Sprintf("%s=%s", constants.OverwriteDefaultInstallationPathEnvVarName, filepath.Join(ts.Dirs.Work, "multi-file")),
+			),
+		)
+		cp.Expect("Updating State Tool to latest version available")
+		cp.Expect("Version updated to", 60*time.Second)
+		cp.ExpectExitCode(0)
+
+		exeExt := ""
+		if runtime.GOOS == "windows" {
+			exeExt = ".exe"
+		}
+		suite.Assert().FileExists(filepath.Join(ts.Dirs.Work, "state"+exeExt), "Transitional state tool script does not exist.")
+		suite.FileExists(filepath.Join(ts.Dirs.Work, "multi-file", "state-svc"+exeExt))
+		suite.FileExists(filepath.Join(ts.Dirs.Work, "multi-file", "state-tray"+exeExt))
+	})
+
+}
+
+// TestAutoUpdateFlow meant to test whether updating from an untagged release-candidate will auto-update to a v29 update forwarding to the new State Tool.
+func (suite *V29TestSuite) TestAutoUpdateFlow() {
```

Comment:
This test seems to be repeating the `Update without tag` test from above. Are they both needed?

Reply:
Yeah.  This one is testing the Auto update flow.  `Update without tag` is updating via `state update`.